librosa/core/audio.py

```diff
@@ -139,6 +139,8 @@ def load(path, sr=22050, mono=True, offset=0.0, duration=None,
 
     except RuntimeError as exc:
         # If soundfile failed, fall back to the audioread loader
+        if hasattr(path, 'name'):
```

Comment:
If a file cannot be parsed by `soundfile`, the resulting error leads to this `audioread` fallback and `audioread will subsequently complain that it expects a string and not a file. This new if check is a hacky way of getting the underlying path from an `open()` call but not all file like objects have a name so this is very fragile!

Probably better to check why `soundfile` failed and handle distinct cases differently? Also, should a warning be logged here to the user because now it's failing silently which is confusing (sometimes loading is super fast, sometimes it's slow, depending on if e.g. `sudo apt install libsndfile1` was run on the user machine or not).

Reply:
> Probably better to check why `soundfile` failed and handle distinct cases differently?

Good point.  In the first cut of this, I was really only concerned with missing codec errors, but soundfile can throw a bunch of typeerrors or valueerrors that are not handled here.  I think it's fine to propagate soundfile errors upstream if we can't compensate for them by falling back on audioread.



> This new if check is a hacky way of getting the underlying path from an `open()` call but not all file like objects have a name so this is very fragile!

I don't think we should do this.  If the user supplies an open file handle, and soundfile can't work with it, we shouldn't try to make audioread do it.  In this case, it should fail noisily.

I think the logic should be that audioread fallback is only triggered if the input is a path string, and otherwise any exceptions get reraised.



> Also, should a warning be logged here to the user because now it's failing silently which is confusing

Yes, excellent point!

Reply:
>if the input is a path string

This is pretty hard to check in a a Python 2 _and_ 3 compatible way.

Reply:
`isinstance(path, six.string_types())` ought to do the trick. :)

Reply:
Ooh, convenient.